Objective-C/CBLQueryResult.mm

```diff
@@ -46,7 +46,9 @@ - (instancetype) initWithResultSet: (CBLQueryResultSet*)rs
 
 
 - (NSUInteger) count {
-    return c4query_columnCount(_rs.c4Query);
+    CBL_LOCK(self) {
```

Comment:
Other uses of C4Query (in CBLQuery.mm) are synchronized with the CBLDatabase, so this should be too, right?